kore/app/check-functions/Main.hs

```diff
@@ -0,0 +1,92 @@
+module Main (main) where
+
+import qualified Data.Text.IO as Text
+import GlobalMain
+import Kore.Attribute.Function (
+    isDeclaredFunction,
+ )
+import Kore.Attribute.Pattern.Function (
+    isFunction,
+ )
+import Kore.Equation (
+    Equation (Equation),
+    right,
+ )
+import Kore.Equation.Sentence (
+    fromSentenceAxiom,
+ )
+import Kore.Internal.TermLike (
+    isFunctionPattern,
+ )
+import Kore.Options (
+    InfoMod,
+    Parser,
+    argument,
+    fullDesc,
+    header,
+    help,
+    metavar,
+    progDesc,
+    str,
+ )
+import Kore.Parser (
+    parseKoreDefinition,
+ )
+import Kore.Syntax.Definition (
+    definitionModules,
+ )
+import Kore.Syntax.Module (
+    Module (..),
+ )
+import Kore.Syntax.Sentence (
+    projectSentenceAxiom,
+    sentenceAxiomAttributes,
+    sentenceAxiomPattern,
+ )
+import Prelude.Kore
+
+-- | modifiers for the command line parser description
+checkerInfoModifiers :: InfoMod options
+checkerInfoModifiers =
+    fullDesc
+        <> progDesc
+            "Checks function definitions in FILE and verifies that for every \
+            \equation in a function definition, the right-hand side of the \
+            \equation is a function pattern."
+        <> header "kore-check-functions - a tool to check function definitions"
+
+newtype KoreCheckerOptions = KoreCheckerOptions
+    { -- | Name for a file containing function definitions to verify.
+      fileName :: FilePath
+    }
+
+parseKoreCheckerOptions :: Parser KoreCheckerOptions
+parseKoreCheckerOptions =
+    KoreCheckerOptions
+        <$> argument
+            str
+            ( metavar "FILE"
+                <> help "Kore source file to check."
+            )
+
+main :: IO ()
+main = do
```

Comment:
This should be similar to https://github.com/kframework/kore/pull/2697

Reply:
Use [Equation.extractEquations](https://github.com/kframework/kore/blob/8cf981dae53f6bd8c6f9cd97d478f9efbd602324/kore/src/Kore/Equation/Registry.hs#L57)